pkg/testhelper/testhelper.go

```diff
@@ -180,6 +189,14 @@ func (obj *ReportObject) AddField(aKey, aValue string) (out *ReportObject) {
 	return obj
 }
 
+func NewNamespacedReportObject(aReason, aType string, isCompliant bool, aNamespace string) (out *ReportObject) {
```

Comment:
This got me thinking, what if we just baked the `ReportObject` into the objects themselves.  Such as like Pod.NewReportObject()?

Reply:
not sure if it could work, the report object is really custom based on the specific processing of the test, not just the content of the base structures: we might need more than one per object or even one mixing base objects. I fell if we add it there would be harder to maintain. In my opinion, the failure reason should feel like writing text annotating the actual tests we are running.